# GEOS-Chem

In [1]:
import xarray as xr

In [2]:
meteo = xr.open_dataset('/uufs/chpc.utah.edu/common/home/u1474799/haskins-group1/data/ExtData/GEOS_0.25x0.3125_NA/GEOS_FP/2017/01/GEOSFP.20170101.A1.025x03125.NA.nc')

In [ ]:
# utah boundary:
# Latitude	37° N to 42° N
# Longitude	109°3′ W to 114°3′ W

# https://en.wikipedia.org/wiki/Utah

In [3]:
rightbottom = meteo.sel(lat = 37, lon = -109.3, method = 'nearest')
leftbottom = meteo.sel(lat = 37, lon = -114.3, method = 'nearest')
righttop = meteo.sel(lat = 42, lon = -114.3, method = 'nearest')
lefttop = meteo.sel(lat = 42, lon = -109.3, method = 'nearest')

In [4]:
rightbottom.lat.values, rightbottom.lon.values

(array(37., dtype=float32), array(-109.375, dtype=float32))

In [5]:
leftbottom.lat.values, leftbottom.lon.values

(array(37., dtype=float32), array(-114.375, dtype=float32))

In [6]:
righttop.lat.values, righttop.lon.values

(array(42., dtype=float32), array(-114.375, dtype=float32))

In [7]:
lefttop.lat.values, lefttop.lon.values

(array(42., dtype=float32), array(-109.375, dtype=float32))

**Therefore, if we are going to use the GEOS-Chem, the grid lat is 37 - 42, and the grid lon is -109.375 to 114.375.**

# ERA5 hourly analysis

data link: https://cds.climate.copernicus.eu/datasets/reanalysis-era5-single-levels?tab=overview.

In [ ]:
netds = xr.open_dataset("/uufs/chpc.utah.edu/common/home/u1474799/szhao/Algorithm/meteo_replacement/ERA_real_check.nc")

I've checked the lat and lon, for the utah, we can use:  

+ lat: 37 - 42 (stride: 0.25)

+ lon: -109 - -114.  (stride: 0.25)

Grid resolution: 0.25 * 0.25.

**If we are going to consider air quality, then use the 0.25 * 0.3125 grid resolution eventually.**